x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -741,10 +750,18 @@ export class TaskRunner<
           if (isAlertSavedObjectNotFoundError(err, ruleId)) {
             this.logger.debug(message);
           } else {
-            this.logger.error(message, {
-              tags: [this.ruleType.id, ruleId, 'rule-run-failed'],
-              error: { stack_trace: err.stack },
-            });
+            if (this.stackTraceLog) {
```

Comment:
WDYT of doing something like:

```
const error = this.stackTraceLog ? this.stackTraceLog.message : err;
const stack = this.stackTraceLog ? this.stackTraceLog.stackTrace : err.stack;
const message = `Executing Rule ${spaceId}:${this.ruleType.id}:${ruleId} has resulted in Error: ${getEsErrorMessage(error)} - ${stack ?? ''}`;
```

and then just logging as before

```
this.logger.error(message, {
  tags: [this.ruleType.id, ruleId, 'rule-run-failed'],
  error: { stack_trace: err.stack },
});
```

That way we always get the `Executing Rule ... has resulted in Error` message logged regardless of where the error originated from? It's nice to have something consistent to search for in the logs when looking for errors

Reply:
Yes I like that, I will update it. Thanks!

Reply:
Resolved in https://github.com/elastic/kibana/pull/139721/commits/fe2ab3bf3e1ee4e72131ed85f0dec020f0a3e9f5